napalm_nxos/nxos.py

```diff
@@ -734,7 +876,10 @@ def get_mac_address_table(self):
     def get_snmp_information(self):
         snmp_information = {}
         snmp_command = 'show running-config'
-        snmp_raw_output = self.cli([snmp_command]).get(snmp_command, '')
+        if self.ssh_connection:
```

Comment:
I think this pattern is going to be a problem, it's very redundant and people if going to forget about it. I would certainly do ;)

I suggest hiding the logic inside `self.cli()` or any equivalent method.